### PR TITLE
Remove AP_InertialNav get_filter_status

### PR DESCRIPTION
**Pull request by peterbarker**
This just passed through the EKF filter status, but lost the bool result code...

This is based on top of another PR; it should be merged first.

### CODE REVIEW

File: ArduCopter/events.cpp [hunk 1]
```diff
@@ -206,7 +206,11 @@ void Copter::failsafe_terrain_on_event()
 void Copter::gpsglitch_check()
 {
     // get filter status
-    nav_filter_status filt_status = inertial_nav.get_filter_status();
+    nav_filter_status filt_status;
+    if (!ahrs.get_filter_status(filt_status)) {
+        // Errr.. maybe?
```
Comment (OXINARF) [inline ArduCopter/events.cpp]: Seems correct, we can't know if we can't get it so we can only return.

File: libraries/AC_PrecLand/AC_PrecLand.cpp [hunk 2]
```diff
@@ -162,7 +162,13 @@ void AC_PrecLand::update(float rangefinder_alt_cm, bool rangefinder_alt_valid)
     _ahrs.getCorrectedDeltaVelocityNED(inertial_data_newest.correctedVehicleDeltaVelocityNED, inertial_data_newest.dt);
     inertial_data_newest.Tbn = _ahrs.get_rotation_body_to_ned();
     inertial_data_newest.inertialNavVelocity = _inav.get_velocity()*0.01f;
-    inertial_data_newest.inertialNavVelocityValid = _inav.get_filter_status().flags.horiz_vel;
+    inertial_data_newest.inertialNavVelocityValid = false;
+    if (AP::ahrs().have_inertial_nav()) {
+        Vector3f vel;
+        if (AP::ahrs().get_velocity_NED(vel)) {
+            inertial_data_newest.inertialNavVelocityValid = true;
```
Comment (OXINARF) [inline libraries/AC_PrecLand/AC_PrecLand.cpp]: I'd prefer that the get_velocity above was replaced too, it would make this code look OK.

**Comment by peterbarker** [comment]
On Wed, 4 Apr 2018, Francisco Ferreira wrote:

> @OXINARF commented on this pull request.
> 
> Same comment as in the other PR regarding removing methods from
> AP_InertialNav.

Same response :-P

> In ArduCopter/events.cpp:
> 
> > @@ -206,7 +206,11 @@ void Copter::failsafe_terrain_on_event()
>  void Copter::gpsglitch_check()
>  {
>      // get filter status
> -    nav_filter_status filt_status = inertial_nav.get_filter_status();
> +    nav_filter_status filt_status;
> +    if (!ahrs.get_filter_status(filt_status)) {
> +        // Errr.. maybe?
> 
> Seems correct, we can't know if we can't get it so we can only return.

We could return bool to indicate if the check was successful, at least - 
even if the caller's going to ignore it.  Need to do so in a way that 
doesn't make the caller thing we're returning true if a glitch was 
detected.

>      _ahrs.getCorrectedDeltaVelocityNED(inertial_data_newest.correctedVehicl
> eDeltaVelocityNED, inertial_data_newest.dt);
>      inertial_data_newest.Tbn = _ahrs.get_rotation_body_to_ned();
>      inertial_data_newest.inertialNavVelocity = _inav.get_velocity()*0.01f;
> -    inertial_data_newest.inertialNavVelocityValid = _inav.get_filter_status
> ().flags.horiz_vel;
> +    inertial_data_newest.inertialNavVelocityValid = false;
> +    if (AP::ahrs().have_inertial_nav()) {
> +        Vector3f vel;
> +        if (AP::ahrs().get_velocity_NED(vel)) {
> +            inertial_data_newest.inertialNavVelocityValid = true;
> 
> I'd prefer that the get_velocity above was replaced too, it would make this
> code look OK.

get_velocity is going to die in later patches...

.. and yes, this code will look much better at that point.  I'm kind of 
curious about having to call have_inertial_nav() here.  The comment on 
that function seemed to indicate so, but I have my doubts it should 
generally be required...

**Comment by peterbarker** [comment]
Too old.  Too many conflicts.  Faster to redo.

**Pull request closed without merging**